include/wlr/types/wlr_foreign_toplevel_management_v1.h

```diff
@@ -0,0 +1,120 @@
+/*
+ * This an unstable interface of wlroots. No guarantees are made regarding the
+ * future consistency of this API.
+ */
+#ifndef WLR_USE_UNSTABLE
+#error "Add -DWLR_USE_UNSTABLE to enable unstable wlroots features"
+#endif
+
+#ifndef WLR_TYPES_WLR_FOREIGN_TOPLEVEL_MANAGEMENT_V1_H
+#define WLR_TYPES_WLR_FOREIGN_TOPLEVEL_MANAGEMENT_V1_H
+
+#include <wayland-server.h>
+#include <wlr/types/wlr_output.h>
+
+struct wlr_foreign_toplevel_manager_v1 {
+	struct wl_event_loop *event_loop;
```

Comment:
Should I keep the pointer to the toplevel manager or drop it(it is used only for the event loop, which as you say should  come from the resource/client)?

Reply:
Dropping it is the reason why I'm suggesting to use `wl_client_get_display` :)

Reply:
okay, will do it sometime later today

Reply:
This won't work very well. Where do I get the client for `wl_client_get_display()`? The resources list might be empty.

Reply:
Hmm, actually we can do this: Use the client from the first resource (though I'm not sure how to access the first element in a `wl_list`, I'd be helpful if somebody hints me). If no clients, don't send done, because no client will receive it anyway. Is this fine?

Reply:
Okay, my bad, let's just keep it that way. I thought it was in a request handler, but it's not. So it's better to keep things simple instead of having this minor improvement.

For the record, here's how you access the first element of a list:

```c
struct thing *thing = wl_container_of(parent->children.next, thing, link);
```